bouncer/src/repo/core/handler/fileservice/repo_file_handler_fs.cpp

```diff
@@ -136,8 +136,17 @@ std::string FSFileHandler::uploadFile(
 	bool failed;
 	do {
 		std::ofstream outs(path.string(), std::ios::out | std::ios::binary);
-		outs.write((char*)bin.data(), bin.size());
-		outs.close();
+
+		try {
+			outs.write((char*)bin.data(), bin.size());
+			outs.close();
+		}
+		catch (const boost::filesystem::filesystem_error& e)
```

Comment:
Does this work? 

Considering we're using the std ofstream here, I would've never dreamt of it throwing a boost error - or am I missing something?

Reply:
We have changed the file writing class from std::ofstream to boost::filesystem::ofsream for better error handling and log writing on getting an exception.

Reply:
Have you tested it to see if it actually throws an exception? Looking at the boost documentation it claims it has no semantic differences, so it's not going to be doing anything that std library version doesn't do.

![image](https://github.com/3drepo/3drepobouncer/assets/11945337/71fa91aa-407c-4809-9560-48d502cd25a8)

Reply:
@carmenfan We can do the exception handling but it is not giving proper error in a proper readable format, so this is not useful.